internal/app/machined/internal/phase/upgrade/leave_etcd.go

```diff
@@ -55,6 +55,10 @@ func (task *LeaveEtcd) standard(r runtime.Runtime) (err error) {
 		return err
 	}
 
+	if len(resp.Members) == 1 {
+		return fmt.Errorf("only 1 etcd member found. assuming this is not an HA setup and refusing to leave cluster")
+	}
+
```

Comment:
We have a `ValidateForUpgrade` function in `internal/pkg/etcd` that is responsible for all the safety check for etcd. We should probably throw this in there.

Reply:
ah, good call. fixed in amended commit.